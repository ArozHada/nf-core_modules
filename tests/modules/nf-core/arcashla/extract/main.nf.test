nextflow_process {

    name "Test Process ARCASHLA_EXTRACT"
    script "modules/nf-core/arcashla/extract/main.nf"
    process "ARCASHLA_EXTRACT"
    config "tests/modules/nf-core/arcashla/extract/nextflow.config"
    tag "arcashla_extract"

    test("Human Illumina PE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/arcashla/versions.yml")).match("human_pe_bam") },
                { assert new File("$outputDir/arcashla/test.paired_end.sorted.extracted.1.fq.gz").exists() },
                { assert new File("$outputDir/arcashla/test.paired_end.sorted.extracted.2.fq.gz").exists() },
                { assert new File("$outputDir/arcashla/test.log").exists() }
            )
        }

    }

    test("Human Illumina SE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test_single_end', single_end:true ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test3_single_end_markduplicates_sorted_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/arcashla/versions.yml")).match("human_se_bam") },
                { assert new File("$outputDir/arcashla/test3.single_end.markduplicates.sorted.extracted.fq.gz").exists() },
                { assert new File("$outputDir/arcashla/test_single_end.log").exists() }
            )
        }

    }

}
