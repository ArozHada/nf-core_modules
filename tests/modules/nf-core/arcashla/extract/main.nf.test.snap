{
    "human_pe_bam": {
        "content": [
            "versions.yml:md5,bad529c7a165615b9641e7780cc13160"
        ],
        "timestamp": "2023-04-06T02:02:46+0000"
    },
    "human_se_bam": {
        "content": [
            "versions.yml:md5,bad529c7a165615b9641e7780cc13160"
        ],
        "timestamp": "2023-04-06T02:02:46+0000"
    }
}